### PR TITLE
Permissions & docker quota tool

### PR DESCRIPTION
**Pull request by thomasyu888**
Add unit tests for set permissions
Add in docker quota limiter

### CODE REVIEW

File: challengeutils/helpers.py [hunk 1]
```diff
@@ -58,3 +58,35 @@ def create_team_wikis(syn, synid, templateid, tracker_table_synid):
             #syn.sendMessage(i[])
             #Store copied synId to tracking table
             syn.store(synapseclient.Table(tracker_table_synid,[[wiki_copy[templateid],i['teamId']]]))
+
+
+def kill_docker_submission_over_quota(syn, evaluation_id, quota=None):
```
Comment (kdaily) [inline challengeutils/helpers.py]: this only works with the workflow hook though?

File: challengeutils/permissions.py [hunk 2]
```diff
@@ -27,11 +27,12 @@ def set_entity_permissions(syn, entity, principalid=None, permission_level="view
     """
     Convenience function to set ACL on an entity for a user or team based on
     permission levels (view, download...)
-
+    
     Args:
+        syn: Synapse object
         entity: An Entity or Synapse ID to lookup
         principalid: Identifier of a user or group (defaults to PUBLIC users)
```
Comment (kdaily) [inline challengeutils/permissions.py]: says defaults to public but I don't think this is true

File: challengeutils/permissions.py [hunk 3]
```diff
@@ -42,8 +43,14 @@ def set_entity_permissions(syn, entity, principalid=None, permission_level="view
...
-    assert permission_level in permission_level_mapping.keys()
-    assert principalid is not None
-    entity = syn.get(entity)
-    syn.setPermissions(entity, principalId=principalid, accessType=permission_level_mapping[permission_level])
+                                'admin':admin,
+                                'delete':[]}
+    if permission_level not in permission_level_mapping.keys():
+        raise ValueError("permission_level must be one of these: {0}".format(', '.join(permission_level_mapping.keys())))
+
+    if principalid is None:
```
Comment (kdaily) [inline challengeutils/permissions.py]: conflicts docstring

File: challengeutils/permissions.py [hunk 4]
```diff
@@ -42,8 +43,14 @@ def set_entity_permissions(syn, entity, principalid=None, permission_level="view
...
+        raise ValueError("principalid must not be None")
+
+    entityid = synapseclient.utils.id_of(entity)
+    syn.setPermissions(entityid, principalId=principalid, accessType=permission_level_mapping[permission_level])
```
Comment (kdaily) [inline challengeutils/permissions.py]: long line, reformat

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+sudo: required
+language: python
+script:
+  - python setup.py install
+  - pytest -vs tests
+
+before_install:
+  - pip install pandas synapseclient pycrypto
```
Comment (kdaily) [inline .travis.yml]: why pycrypto requirement?

File: challengeutils/helpers.py (hunk 1 above)
Comment (thomasyu888) [reply challengeutils/helpers.py]: Yes. This only works with the hook.

**Pull request merged**